pkgs/top-level/all-packages.nix

```
@@ -14344,7 +14344,7 @@ with pkgs;
     gconf = gnome2.GConf;
   };
 
-  tinycc = callPackage ../development/compilers/tinycc { };
+  tinycc = darwin.apple_sdk_11_0.callPackage ../development/compilers/tinycc { };
```

```suggestion
  tinycc = callPackage ../development/compilers/tinycc {
    stdenv = if stdenv.isDarwin then darwin.apple_sdk_11_0.stdenv else stdenv;
  };
```

I'm afraid that it's a good practice to build everything on `darwin.apple_sdk_11_0.stdenv` even when you're not on darwin.

---

This is what go1.18 is doing, also note the platform doc in this ripgrep:

```
λ rg "darwin.apple_sdk_11_0.callPackage"
doc/stdenv/platform-notes.chapter.md
66:  `apple_sdk` in your derivation and use `pkgs.darwin.apple_sdk_11_0.callPackage` instead of `pkgs.callPackage`. On Linux, this will
67:  have the same effect as `pkgs.callPackage`, so you can use `pkgs.darwin.apple_sdk_11_0.callPackage` regardless of platform.

pkgs/top-level/all-packages.nix
13457:  go_1_18 = darwin.apple_sdk_11_0.callPackage ../development/compilers/go/1.18.nix {
21829:  buildGo118Package = darwin.apple_sdk_11_0.callPackage ../development/go-packages/generic {
21840:  buildGo118Module = darwin.apple_sdk_11_0.callPackage ../development/go-modules/generic {
```

---

`callPackage` and `stdenv` under `pkgs.darwin.apple_sdk_11_0.callPackage` is quite new and there is no standard way of packaging with it. Maybe I should create an issue and discuss about the best practice? Thanks for letting me know about the document!

---

yep. ideally we can just get this merged to unblock deno & have a discussion elsewhere.

The new nixpkgs-architecture team might be interested https://github.com/nixpkgs-architecture